File: Chapter13/VB/CommonTypes/PricedItem.vb
<Serializable()> _
Public Class PricedItem

  Private _name As String
  Public Property Name() As String
    Get
      Return _name
    End Get
    Set(ByVal Value As String)
      _name = Value
    End Set
  End Property

  Private _cost As Decimal
  Public Property Cost() As Decimal
    Get
      Return _cost
    End Get
    Set(ByVal Value As Decimal)
      _cost = Value
    End Set
  End Property

  Public Overrides Function ToString() As String
    Return _name
  End Function

  ' public default constructor is required in serializable classes
  Public Sub New()
  End Sub

  Public Sub New(ByVal theName As String, ByVal theCost As Decimal)
    _name = theName
    _cost = theCost
  End Sub
End Class

File: Chapter13/C#/Common Types/PricedItem.cs
using System;

namespace Asap.Cars.CommonTypes
{
	/// <summary>
	/// Summary description for PricedItem.
	/// </summary>
  [Serializable]
  public class PricedItem
	{
    string name;
    public string Name
    {
      get {return name;}
      set {name = value;}
    }

    decimal cost;
    public decimal Cost
    {
      get {return cost;}
      set {cost = value;}
    }

    public override string ToString()
    {
      return name;
    }

    // public default constructor is required in serializable classes
    public PricedItem()
    {
    }
    
    public PricedItem(string theName, decimal theCost)
		{
      name = theName;
      cost = theCost;
		}
	}
}
